### ISSUE TITLE
Cannot create custom shadow for LayoutInflater

### ISSUE THREAD

**Issue by pimterry** [robolectric/robolectric#1459]
I want to create my own custom LayoutInflater to override the layout used in certain cases (as in #1372). I'm hitting a different issue though, and despite using the solution suggested on that ticket I'm not getting anywhere.. 

My layout inflater shadow code looks like:

```
@Implements(value = LayoutInflater.class)
public class OverridingShadowInflater {

    @RealObject
    private LayoutInflater realInflater;

    @Implementation
    public View inflate(int resource, ViewGroup root, boolean attachToRoot) {
        if (resource == R.layout.template_index_card) resource = R.layout.template_index_block;

        return Robolectric.directlyOn(realInflater, LayoutInflater.class).inflate(resource, root, attachToRoot);
    }
}
```

When I run this though, I get: 

```java
error instantiating android.view.LayoutInflater
java.lang.RuntimeException: error instantiating android.view.LayoutInflater
	at org.robolectric.internal.ReflectionHelpers.callConstructorReflectively(ReflectionHelpers.java:104)
	at org.robolectric.bytecode.RobolectricInternals.newInstance(RobolectricInternals.java:20)
	at org.robolectric.bytecode.RobolectricInternals.directlyOn(RobolectricInternals.java:24)
	at org.robolectric.Robolectric.directlyOn(Robolectric.java:368)
	at myproject.helpers.OverridingShadowInflater.inflate(OverridingShadowInflater.java:25)
	at android.view.LayoutInflater.inflate(LayoutInflater.java)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:353)
        .....
Caused by: java.lang.InstantiationException
	at sun.reflect.InstantiationExceptionConstructorAccessorImpl.newInstance(InstantiationExceptionConstructorAccessorImpl.java:48)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:408)
	at org.robolectric.internal.ReflectionHelpers.callConstructorReflectively(ReflectionHelpers.java:102)
	at org.robolectric.bytecode.RobolectricInternals.newInstance(RobolectricInternals.java:20)
	at org.robolectric.bytecode.RobolectricInternals.directlyOn(RobolectricInternals.java:24)
	at org.robolectric.Robolectric.directlyOn(Robolectric.java:368)
	at myproject.helpers.OverridingShadowInflater.inflate(OverridingShadowInflater.java:25)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.robolectric.bytecode.ShadowWrangler$ShadowMethodPlan.run(ShadowWrangler.java:507)
	at android.view.LayoutInflater.inflate(LayoutInflater.java)
	at android.view.LayoutInflater.$$robo$$LayoutInflater_1d1f_inflate(LayoutInflater.java:353)
	at android.view.LayoutInflater.inflate(LayoutInflater.java)
	...
```

There's no more detail inside the InstantiationException. Any ideas?

Alternatively, can you suggest any better way to have my tests use a slightly different layout to my real code? One of my layouts uses a CardView, which isn't supported, so I'm intending to just swap that layout for a test-only version that does something simpler, but I'm finding that remarkably difficult.

**Comment by erd** [comment]
> One of my layouts uses a CardView

CardView from the support library should work out of the box on 3.0, for what it's worth.

**Comment by jaredsburrows** [comment]
@erd  What is the ETA for when Robolectric 3.0 will be released?

**Comment by erd** [comment]
No ETA yet.

**Comment by jaredsburrows** [comment]
@erd  Since Robolectric 3.0 fixes many issues from 2.4 and its already in the master branch, you should make an official release. This would also prevent(or slow down) issues from Robolectric 2.4(or older) from being created. What other changes are you wanted to add into the 3.0 release?

**Comment by erd** [comment]
This should be resolved by using `RobolectricGradleTestRunner` in 3.0 (it should work in 3.0-rc2).
